bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/handler/DaikinAcUnitHandler.java

```diff
@@ -44,49 +50,93 @@
+            updateStatus(ThingStatus.ONLINE);
+            if (controlInfo != null) {
+                updateState(DaikinBindingConstants.CHANNEL_AC_POWER, controlInfo.power ? OnOffType.ON : OnOffType.OFF);
+                updateTemperatureChannel(DaikinBindingConstants.CHANNEL_AC_TEMP, controlInfo.temp);
+
+                updateState(DaikinBindingConstants.CHANNEL_AC_MODE, new StringType(controlInfo.mode.name()));
+                updateState(DaikinBindingConstants.CHANNEL_AC_FAN_SPEED, new StringType(controlInfo.fanSpeed.name()));
+                updateState(DaikinBindingConstants.CHANNEL_AC_FAN_DIR, new StringType(controlInfo.fanMovement.name()));
+
+                if (!controlInfo.power) {
+                    updateState(DaikinBindingConstants.CHANNEL_AC_HOMEKITMODE,
+                            new StringType(HomekitMode.OFF.getValue()));
+                } else if (controlInfo.mode == Mode.COLD) {
+                    updateState(DaikinBindingConstants.CHANNEL_AC_HOMEKITMODE,
+                            new StringType(HomekitMode.COOL.getValue()));
+                } else if (controlInfo.mode == Mode.HEAT) {
+                    updateState(DaikinBindingConstants.CHANNEL_AC_HOMEKITMODE,
+                            new StringType(HomekitMode.HEAT.getValue()));
+                } else if (controlInfo.mode == Mode.AUTO) {
+                    updateState(DaikinBindingConstants.CHANNEL_AC_HOMEKITMODE,
+                            new StringType(HomekitMode.AUTO.getValue()));
+                }
+
+                updateState(DaikinBindingConstants.CHANNEL_AC_SPECIALMODE,
+                        new StringType(controlInfo.specialMode.name()));
+
+                if (controlInfo.specialMode.isUndefined()) {
+                    updateState(DaikinBindingConstants.CHANNEL_AC_SPECIALMODE_POWERFUL, UnDefType.UNDEF);
+                } else {
+                    updateState(DaikinBindingConstants.CHANNEL_AC_SPECIALMODE_POWERFUL,
+                            controlInfo.specialMode.isPowerfulActive() ? OnOffType.ON : OnOffType.OFF);
+                }
+
             }
-        }
 
-        SensorInfo sensorInfo = webTargets.getSensorInfo();
-        if (sensorInfo != null) {
-            updateTemperatureChannel(DaikinBindingConstants.CHANNEL_INDOOR_TEMP, sensorInfo.indoortemp);
+            SensorInfo sensorInfo = webTargets.getSensorInfo();
+            if (sensorInfo != null) {
+                updateTemperatureChannel(DaikinBindingConstants.CHANNEL_INDOOR_TEMP, sensorInfo.indoortemp);
+
+                updateTemperatureChannel(DaikinBindingConstants.CHANNEL_OUTDOOR_TEMP, sensorInfo.outdoortemp);
 
-            updateTemperatureChannel(DaikinBindingConstants.CHANNEL_OUTDOOR_TEMP, sensorInfo.outdoortemp);
+                if (sensorInfo.indoorhumidity.isPresent()) {
+                    updateState(DaikinBindingConstants.CHANNEL_HUMIDITY,
+                            new DecimalType(sensorInfo.indoorhumidity.get()));
```

Comment:
humidity should use QuantityType with `SmartHomeUnits.PERCENT` units.

Note: Changing this might be outside the scope of this PR so you are free to skip this change.

Reply:
Corrected.